identity/client/src/utility/api/hooks/usePaginatedApiCall.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+
+import { useCallback, useEffect, useState } from "react";
+import useApiCall from "./useApiCall";
+import usePagination, { PageResult } from "./usePagination";
+import { mergeParams } from "./utils";
+
+const usePaginatedApiCall = <T extends { page?: PageResult }, P>(
+  ...props: Parameters<typeof useApiCall<T, P>>
+) => {
+  const { pageParams, page, setPage, setPageSize, ...rest } = usePagination();
+
+  const [pageResult, setPageResult] = useState<Partial<PageResult>>({});
+
+  const [_call, result, reset] = useApiCall<T, P>(...props);
+
+  const call = useCallback(
+    (params: P) => {
+      return _call(
+        mergeParams(params as Record<string, unknown>, pageParams) as P,
+      );
+    },
+    [_call, pageParams],
+  );
+
+  useEffect(() => {
+    if (result.data?.page) {
+      setPageResult(result.data.page);
+    }
+  }, [result.data]);
```

Comment:
❓ Could you clarify why a `useEffect` is used here to handle `result.data.page`, while `usePaginatedApi` manages similar logic without a `useEffect`? Is the difference intentional or could both hooks be unified for consistency?

Reply:
Great catch! You are right, it is not needed here. We don't even need the state, we can just map data types directly in the return. This makes this hook much simpler

I thought state was needed since we handle multiple API calls instead of a single one, but that is already covered by `useApiCall`, no need to duplicate it.